packages/core/src/libraries/hook/index.ts

```diff
@@ -90,12 +85,10 @@ export const createHookLibrary = (queries: Queries) => {
 
         // Trigger web hook and log response
         await got
-          .post(url, {
-            headers: { 'user-agent': 'Logto (https://logto.io)', ...headers },
-            json,
-            retry: { limit: retries ?? 3 },
-            timeout: { request: 10_000 },
-          })
+          .post(
+            url,
+            createHookRequestOptions({ signingKey, payload: json, customHeaders: headers, retries })
```

Comment:
Better keep it inline? If you prefer, could define a sendWebHook method exclusively.  Sth like

`await Promise.all(rows.map(sendWebHook));`

createHookRequestOptions method scope feels unclear.

Reply:
Updated by keeping it inline since the code refactors belong to the next PR about 'testing webhook'.